### PR TITLE
Improve docs for older React Aria hooks

### PR DESCRIPTION
**Pull request by devongovett**
Based on #3264 for code changes.

This updates some older React Aria docs with sections and patterns we have added in newer components. 

* Reuse components rather than passing the output from one hook to another. When a component is reused, its code is in a collapsed section.
* Add a Usage section with more examples of how to use a built component. Content mainly copied from Spectrum docs.
* Update examples to simplify where possible. e.g. reduce the amount of styling code
* Take advantage of states returned from hooks in addition to props rather than computing things from state manually.

One larger change to note is that I merged `useMenuTrigger` and `useMenu` into a single page since you would never use one without the other. The format is slightly different from other docs due to interleaving tutorial style and usage sections. Feedback welcome.

### CODE REVIEW

File: packages/@react-aria/meter/docs/useMeter.mdx [hunk 1]
```diff
@@ -113,8 +113,46 @@ function Meter(props) {
...
+  value={25} />
+```
+
+## Usage
+
+The following examples show how to use the `Meter` component created in the above example.
+
+### Custom value scale
+
+By default, the `value` prop represents the current percentage of progress, as the minimum and maximum values default to 0 and 100, respectively. Alternatively, a different scale can be used by setting the `minValue` and m`axValue` props.
```
Comment (snowystinger) [inline packages/@react-aria/meter/docs/useMeter.mdx:125]:
```suggestion
By default, the `value` prop represents the current percentage of progress, as the minimum and maximum values default to 0 and 100, respectively. Alternatively, a different scale can be used by setting the `minValue` and `maxValue` props.
```

File: packages/@react-aria/tooltip/docs/useTooltipTrigger.mdx [hunk 2]
```diff
@@ -77,7 +77,7 @@ This example implements a Tooltip that renders in an absolute position next to i
 
 Two instances of the example are rendered to demonstrate the behavior of the delay on hover. If you hover over the first button, the tooltip will be shown after a delay. Hovering over the second button shows the tooltip immediately. If you wait for a delay before hovering another element, the delay restarts.
 
-```tsx example
+```tsx example export=true
```
Comment (snowystinger) [inline packages/@react-aria/tooltip/docs/useTooltipTrigger.mdx:80]: should we merge useTooltipTrigger with useTooltip like you did for Menu and MenuTrigger?

File: packages/@react-aria/listbox/docs/useListBox.mdx [hunk 3]
```diff
@@ -333,18 +416,16 @@ For options that have more complex content (e.g. icons, multiple lines of text,
...
 
-**NOTE: listbox options cannot contain interactive content (e.g. buttons, checkboxes, etc.).
-For these cases, see `useGrid` instead.**
+**NOTE: listbox options cannot contain interactive content (e.g. buttons, checkboxes, etc.).**
 
-This example shows how `labelProps` and `descriptionProps` can be applied to child elements of
-the item to apply ARIA properties returned by <TypeLink links={docs.links} type={docs.exports.useOption} />.
-This is done using `React.cloneElement` in this example, but you can use context or other
-approaches for this as well.
+To implement this, we'll update the `Option` component to apply the ARIA properties
```
Comment (snowystinger) [inline packages/@react-aria/listbox/docs/useListBox.mdx:421]: This example should probably be single selection only based on the options in the box

File: packages/@react-aria/listbox/docs/useListBox.mdx [hunk 4]
```diff
@@ -417,6 +497,58 @@ function Option({item, state}) {
...
+## Asynchronous loading
+
+This example uses the [useAsyncList](../react-stately/useAsyncList.html) hook to handle asynchronous loading
+of data from a server. You may additionally want to display a spinner to indicate the loading
+state to the user, or support features like infinite scroll to load more data.
+
+```tsx example
+import {useAsyncList} from '@react-stately/data';
+
+function AsyncLoadingExample() {
```
Comment (snowystinger) [inline packages/@react-aria/listbox/docs/useListBox.mdx:509]: should this example stop loading pretty quickly? it's only going to raticate for me

File: packages/@react-aria/numberfield/docs/useNumberField.mdx [hunk 5]
```diff
@@ -122,16 +123,13 @@ function NumberField(props) {
...
 
-  let {buttonProps: incrementProps} = useButton(incrementButtonProps, incrRef);
-  let {buttonProps: decrementProps} = useButton(decrementButtonProps, decRef);
-
   return (
     <div>
       <label {...labelProps}>{props.label}</label>
       <div {...groupProps}>
-        <button {...decrementProps} ref={decRef}>-</button>
+        <Button {...incrementButtonProps}>-</Button>
```
Comment (snowystinger) [inline packages/@react-aria/numberfield/docs/useNumberField.mdx:130]: these buttons are in reverse ;)

File: packages/@react-aria/searchfield/docs/useSearchField.mdx [hunk 6]
```diff
@@ -106,28 +107,26 @@ function SearchField(props) {
 ## Styling
 
 This example uses CSS to reset the default browser styling for a search field and implement
-custom styles. It also includes a custom clear button, using props
-from <TypeLink links={docs.links} type={docs.exports.useSearchField} />
-and <TypeLink links={buttonDocs.links} type={buttonDocs.exports.useButton} />
-(from @react-aria/button).
+custom styles. It also includes a custom clear button, built with [useButton](useButton.html).
```
Comment (snowystinger) [inline packages/@react-aria/searchfield/docs/useSearchField.mdx:110]: this example jumps a bit in chrome when the clear button appears, maybe set a larger default height?

File: packages/@react-aria/tooltip/docs/useTooltipTrigger.mdx [hunk 7]
```diff
@@ -111,18 +113,62 @@ function Example(props) {
...
-        </Tooltip>
+        <Tooltip state={state} {...tooltipProps}>{props.tooltip}</Tooltip>
       )}
     </span>
   );
 }
 
-<Example />
-<Example />
+<TooltipButton tooltip="Edit">🖊</TooltipButton>
```
Comment (snowystinger) [inline packages/@react-aria/tooltip/docs/useTooltipTrigger.mdx:124]: it's probably always been bad, but these icons are super hard to see in dark mode

File: packages/@react-aria/tooltip/docs/useTooltipTrigger.mdx (hunk 2 above)
Comment (devongovett) [reply packages/@react-aria/tooltip/docs/useTooltipTrigger.mdx:80]: I only see https://react-spectrum.adobe.com/react-aria/useTooltipTrigger.html, there is no useTooltip page. What do you mean?

File: packages/@react-aria/listbox/docs/useListBox.mdx (hunk 4 above)
Comment (devongovett) [reply packages/@react-aria/listbox/docs/useListBox.mdx:509]: It doesn't infinite scroll, no. That would require additional code be added to the component to handle scrolling...

File: packages/@react-aria/tooltip/docs/useTooltipTrigger.mdx (hunk 2 above)
Comment (snowystinger) [reply packages/@react-aria/tooltip/docs/useTooltipTrigger.mdx:80]: I have no idea what I thought I saw, you are correct

**Review by snowystinger — APPROVED** [review]

**Review by dannify — CHANGES_REQUESTED** [review]
tiny typo

**Review by LFDanLu — APPROVED** [review]
LGTM

**Review by dannify — APPROVED** [review]

**Pull request merged**